src/main/scala/org/neo4j/spark/Neo4jOptions.scala

```diff
@@ -120,11 +120,44 @@ class Neo4jOptions(private val parameters: java.util.Map[String, String]) extend
   def initNeo4jRelationshipMetadata(): Neo4jRelationshipMetadata = {
     val source = initNeo4jNodeMetadata(getParameter(RELATIONSHIP_SOURCE_NODE_KEYS, ""),
       getParameter(RELATIONSHIP_SOURCE_LABELS, ""))
+
     val target = initNeo4jNodeMetadata(getParameter(RELATIONSHIP_TARGET_NODE_KEYS, ""),
       getParameter(RELATIONSHIP_TARGET_LABELS, ""))
-    val nodeMap = getParameter(RELATIONSHIP_NODES_MAP, DEFAULT_RELATIONSHIP_NODES_MAP.toString).toString
-      .toBoolean
-    Neo4jRelationshipMetadata(source, target, query.value, nodeMap)
+
+    val nodeMap = getParameter(RELATIONSHIP_NODES_MAP, DEFAULT_RELATIONSHIP_NODES_MAP.toString).toBoolean
+
+    val writeStrategy = RelationshipWriteStrategy
+      .values
+      .find(_.toString == getParameter(RELATIONSHIP_WRITE_STRATEGY, DEFAULT_RELATIONSHIP_WRITE_STRATEGY.toString).toUpperCase)
```

Comment:
```suggestion
    val writeStrategy = RelationshipWriteStrategy.withName(getParameter(RELATIONSHIP_WRITE_STRATEGY, DEFAULT_RELATIONSHIP_WRITE_STRATEGY.toString).toUpperCase)
```

e togli la validazione sotto